addons/payment_stripe/models/payment.py

```diff
@@ -74,8 +74,10 @@ def _add_available_payment_method_types(self, stripe_session_data, tx_values):
             PMT('p24', ['pl'], ['eur', 'pln'], 'punctual'),
         ]
 
+        icon_names = [icon.name.lower() for icon in self.payment_icon_ids]
+        icon_filtered = filter(lambda pmt: pmt.name == 'card' or pmt.name in icon_names, all_payment_method_types)
```

Comment:
As discussed, you need to check that the payment icon exists at all before excluding it.
This is the behavior that we want: Only offer a payment method to the customer if
- Its related payment icon is set on Stripe
- Or if that payment icon couldn't be set because it doesn't exist.

Otherwise, this commit would exclude EPS, Giropay, and P24 from `icon_filtered` in currently existing databases until an upgrade of the module `payment` is executed, which we can't rely upon nor even ask.

(Remember that your tests are run on a fresh database! :wink: )

Reply:
Done

Reply:
To do so, I query all payment.icon records at some point. I did so because:

- I make the hypothesis there won't be many
- Alternatives include as many database query as there are local payment methods (may be acceptable ?) (using ilike)
- Becoming case sensitive because in operator is case sensitive. The issue may arise if a customer already has created an icon for eps by example and didn't use the same case as us and that will certainly conduct him to our support.